Automattic/pocket-casts-android
# Clean up Firebase analytics

## Description

This change cleans up the Firebase analytics and leaves only the essential metrics. It uses the AnalyticsTracker, so the events only need to be added once instead of having Tracks and Firebase.

## Testing Instructions
1. Open the discover section
2. ✅ Verify `discover_list_impression` is called with a `list_id`
3. Tap a paid placement
4. ✅ Verify `discover_list_podcast_tap` is called with a `list_id` and `podcast_uuid`
5. Subscribe to the podcast
6. ✅ Verify `discover_list_podcast_subscribe` is called with `list_id` and `podcast_uuid`
7. Tap play on an episode
8. ✅ Verify `discover_list_episode_play` is called with `list_id` and `podcast_uuid`
